File: apps/apple/Packages/AtelierShared/Sources/AtelierDesign/AtelierTheme.swift
import AtelierCore
import SwiftUI

public enum AtelierPalette {
    public static let paper = Color(red: 0.98, green: 0.96, blue: 0.91)
    public static let ink = Color(red: 0.16, green: 0.15, blue: 0.14)
    public static let coral = Color(red: 0.91, green: 0.35, blue: 0.31)
    public static let cyan = Color(red: 0.20, green: 0.66, blue: 0.72)
    public static let yellow = Color(red: 0.96, green: 0.76, blue: 0.25)

    public static func accent(for product: AppProduct) -> Color {
        switch product {
        case .atelier, .mail: coral
        case .notes: yellow
        case .calendar: cyan
        case .tasks: Color(red: 0.35, green: 0.59, blue: 0.35)
        }
    }
}

public struct AtelierCardModifier: ViewModifier {
    public init() {}

    public func body(content: Content) -> some View {
        content
            .padding()
            .background(.regularMaterial, in: .rect(cornerRadius: 16))
    }
}

public extension View {
    func atelierCard() -> some View {
        modifier(AtelierCardModifier())
    }
}

File: apps/android/design/src/main/kotlin/diy/atelier/design/AtelierTheme.kt
package diy.atelier.design

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.ColorScheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import diy.atelier.core.AppProduct

object AtelierColors {
    val Paper = Color(0xFFF9F4E8)
    val Ink = Color(0xFF292623)
    val Coral = Color(0xFFE8584F)
    val Cyan = Color(0xFF35A8B8)
    val Yellow = Color(0xFFF5C241)
    val Green = Color(0xFF599657)

    fun accent(product: AppProduct): Color = when (product) {
        AppProduct.ATELIER, AppProduct.MAIL -> Coral
        AppProduct.NOTES -> Yellow
        AppProduct.CALENDAR -> Cyan
        AppProduct.TASKS -> Green
    }
}

@Composable
fun AtelierTheme(
    product: AppProduct,
    darkTheme: Boolean = isSystemInDarkTheme(),
    content: @Composable () -> Unit,
) {
    val accent = AtelierColors.accent(product)
    val colors: ColorScheme = if (darkTheme) {
        darkColorScheme(primary = accent, tertiary = AtelierColors.Cyan)
    } else {
        lightColorScheme(
            primary = accent,
            secondary = AtelierColors.Cyan,
            tertiary = AtelierColors.Yellow,
            background = AtelierColors.Paper,
            onBackground = AtelierColors.Ink,
            surface = AtelierColors.Paper,
            onSurface = AtelierColors.Ink,
        )
    }
    MaterialTheme(colorScheme = colors, content = content)
}
